cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaMetadataStateManagerMockTest.java

```diff
@@ -0,0 +1,657 @@
+                kafka,
+                List.of(BROKERS, CONTROLLERS),
+                createKafkaCluster(KAFKA),
+                CLUSTER_CA,
+                CLIENTS_CA);
+
+        KafkaStatus status = new KafkaStatus();
+
+        Checkpoint async = context.checkpoint();
+        kr.reconcile(status, Clock.systemUTC()).onComplete(res -> context.verify(() -> {
+            assertEquals(status.getKafkaMetadataState(), ZooKeeper.name());
+            async.flag();
+        }));
+    }
+
+    /**
+     * Tests warning in Kafka status if wrong annotation applied in KRaft State.
+     *
+     * @param context Test Context
+     */
+    @Test
+    public void testWarningIfWrongAnnotationAppliedInKraftState(VertxTestContext context) {
+
+        Kafka kafka = new KafkaBuilder(KAFKA)
+                .editMetadata()
+                    .addToAnnotations(Annotations.ANNO_STRIMZI_IO_KRAFT, "rollback")
+                .endMetadata()
+                .withNewStatus()
+                    .withKafkaMetadataState(KRaft.name())
+                .endStatus()
+                .build();
+
+        ResourceOperatorSupplier supplier = ResourceUtils.supplierWithMocks(false);
+
+        SecretOperator mockSecretOps = supplier.secretOperations;
+        Secret secret = new Secret();
+        when(mockSecretOps.getAsync(eq(NAMESPACE), eq(KafkaResources.clusterCaCertificateSecretName(CLUSTER_NAME)))).thenReturn(Future.succeededFuture(secret));
+        when(mockSecretOps.getAsync(eq(NAMESPACE), eq(KafkaResources.secretName(CLUSTER_NAME)))).thenReturn(Future.succeededFuture(secret));
+
+        MockKafkaReconciler kr = new MockKafkaReconciler(
+                new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME),
+                vertx,
+                CONFIG,
+                supplier,
+                new PlatformFeaturesAvailability(false, KUBERNETES_VERSION),
+                kafka,
+                List.of(BROKERS, CONTROLLERS),
+                createKafkaCluster(KAFKA),
+                CLUSTER_CA,
+                CLIENTS_CA);
+
+        KafkaStatus status = new KafkaStatus();
+
+        Checkpoint async = context.checkpoint();
+        kr.reconcile(status, Clock.systemUTC()).onComplete(res -> context.verify(() -> {
+            assertTrue(status.getConditions().stream().anyMatch(condition -> "KafkaMetadataStateWarning".equals(condition.getReason())));
+            assertEquals(status.getConditions().get(2).getMessage(), "The strimzi.io/kraft annotation can't be set to migration, rollback or disabled values because the cluster is already KRaft.");
+            assertEquals(status.getKafkaMetadataState(), KRaft.name());
+            async.flag();
+        }));
+    }
+
+    /**
+     * Tests warning in Kafka status if wrong annotation applied in PreKRaft State.
+     *
+     * @param context Test Context
+     */
+    @Test
+    public void testWarningIfWrongAnnotationAppliedInPreKraftState(VertxTestContext context) {
```

Comment:
I think we could remove this test, because `PreKRaft` is not a stable state. It's difficult user changes the annotation on fly.